### ISSUE TITLE
🛒 Shopify Integration

### ISSUE THREAD

**Issue by itsNintu** [webstudio-is/webstudio#2649]
This feature will allow users to integrate their Shopify store with their Webstudio site, offering a unified platform for both web content and e-commerce operations. Users can easily display products, manage them, and process transactions directly on their Webstudio website.

**Comment by MKTempresas** [comment]
I can't understand why an Open Source project involves with closed platform as Shopify instead of WooCommerce or anothers.... just as the CMS functionality!

**Comment by kof** [comment]
@MKTempresas What's there not to understand? Shopify is big and popular, users will ask for that integration first of all. Other integration will follow too

**Comment by kilwaFaraj** [comment]
do you mean you are working on something like https://www.shopyflow.com/?

**Comment by bretoreta** [comment]
> I can't understand why an Open Source project involves with closed platform as Shopify instead of WooCommerce or anothers.... just as the CMS functionality!

I agree with @MKTempresas on this, why not just support open-source alternatives like WooCommerce instead of Shopify

**Comment by TrySound** [comment]
> I can't understand why an Open Source project involves with closed platform as Shopify instead of WooCommerce or anothers.... just as the CMS functionality!

Why "instead" if we can support them all?

**Comment by kof** [comment]
We are not going to build a specialized integration for a single platform. We are building primitives that allow you to build an integration yourself and the integration becomes a template. Community can make 10 shopify templates or 10 WooCommerce templates that would have a different UX.

What we do is supporting all necessary patterns to make the integration great.

**Comment by airntam** [comment]
This is a smart move. Just because we are open source nerds doesn't mean others we may work for are. When a client has an existing Shopify site with 1,500 products tightly integrated with fulfilment and inventory systems, you can't just say "Hey, let's move this into WooCommerce because it's open source".

Having far too many WooCommerce based projects under the belt, I'm personally growing weary of the process. Out of the box, it's limiting for more specialized e-comm setups. Then you're talking plugins or custom dev which means perpetual maintenance and responsibility. 

The lure of Shopify and other SaaS platforms offloads some of that responsibility from developers/designers. If something goes wrong server side, it's Shopify's problem to fix.

Loving this idea of Webstudio purely as the visual side of things. Decoupling the design from the ecommerce backend seems the best of both worlds. 

I'm planning to spend the next year moving all of these tiny WordPress sites I'm responsible for maintaining over to WebStudio. When the Shopify connection is finished and vetted, I'm excited to explore how that will change the game.

I have no idea how I discovered Webstudio but I've not been this excited about web tech for a very long time. It has rekindled a love for designing from the CSS up!

**Comment by kof** [comment]
@airntam very glad to hear it also happy to chat with you more on discord

**Comment by markodraganic** [comment]
I would love to see this integration! It's basically what I was praying for a 2-3 years now.👏 It's time to move on from WP and Woo 😁

**Comment by bretoreta** [comment]
Shopify is just popular in the US and EU, what about Africa? Y'all don't care about that?

**Comment by kof** [comment]
@bretoreta we care, but we have limited resources. We are not building a particular tight integration, we are building a system, so that anyone can integrate any backend. It's just a matter of time until every possible other system is supported by  others and templates provided. We on the core team will be focused on a most popular one.

**Comment by bretoreta** [comment]
ok.

**Comment by kof** [comment]
This ux gets a lot of praise, we should probably look into replicating this ux in the template https://x.com/tobi/status/1856462644674240725

**Comment by erickngimba** [comment]
@kof , Hi .. I'm not really a top notch developer ... But If possible take a look at these guys "Surecarts" they're using same tech as shopify and IMO , I think they are really good than woocommerce. Their platform is working on wordpress currently and they've got so many great reviews.  .. https://surecart.com/ , maybe we can find a way to integrate with them as well?

Thanks!

**Comment by o-be-one** [comment]
I was going to say talk about SureCart like @erickngimba ! You guys must keep a look at SureCart as they intend to be stand-alone for next year, so no more linked to WordPress only. They feel inspired by Shopify and would like to provide some ways to integrate in any other website. Definitively a banger, considering number of customers they already have.

**Comment by aquiveal** [comment]
What's the timeline for the system? Is there a way to create an e-commerce site right now, given that Webstudio is CMS-agnostic? What needs to be implemented to make it e-commerce compatible?

**Comment by aquiveal** [comment]
> [@bretoreta](https://github.com/bretoreta) we care, but we have limited resources. We are not building a particular tight integration, we are building a system, so that anyone can integrate any backend. It's just a matter of time until every possible other system is supported by others and templates provided. We on the core team will be focused on a most popular one.

I see the 'Supabase' and 'Firebase' templates are HTML modules with embedded JavaScript that interacts with the respective services and manipulates the DOM. Will e-commerce integration involve similar front-end JavaScript page templates, or will there be changes to Webstudio's back end?

**Comment by SaveCoHub** [comment]
How far off are we from seeing Headless Shopify integration via Webstudio? An approximation will do.

**Comment by kof** [comment]
@SaveCoHub This year.

**Comment by turbopilotstudio** [comment]
I’m really looking forward to this integration! I plan to implement this with WooCommerce, so I hope it’s plug-and-play. Any updates on when the Shopify integration will be available?

**Comment by arjenvdbijl-ai** [comment]
@kof Any update on the ecommerce update for webstudio.is. Will it before black friday?

**Comment by youcedom** [comment]
Hi team, is the Shopify/ecommerce integration still being worked on or prioritized?

Just checking before making some decisions on our end. Thank you for this!

**Comment by canyild** [comment]
Also waiting on more info on this point, would be a game changer!
